pkg/distributor/distributor_test.go

```diff
@@ -3772,16 +3796,21 @@ func prepare(t testing.TB, cfg prepConfig) ([]*Distributor, []*mockIngester, []*
 		rf = 3
 	}
 
+	ingestersHeartbeatTimeout := 60 * time.Minute
 	ingestersRing, err := ring.New(ring.Config{
 		KVStore: kv.Config{
 			Mock: kvStore,
 		},
-		HeartbeatTimeout:     60 * time.Minute,
+		HeartbeatTimeout:     ingestersHeartbeatTimeout,
 		ReplicationFactor:    rf,
 		ZoneAwarenessEnabled: cfg.totalZones() > 1,
 	}, ingester.IngesterRingKey, ingester.IngesterRingKey, logger, nil)
 	require.NoError(t, err)
-	require.NoError(t, services.StartAndAwaitRunning(context.Background(), ingestersRing))
+	require.NoError(t, services.StartAndAwaitRunning(ctx, ingestersRing))
+	t.Cleanup(func() {
+		// The ring client doesn't stop quickly, so we don't wait.
+		ingestersRing.StopAsync()
+	})
```

Comment:
can this be?
```suggestion
  // The ring client doesn't stop quickly, so we don't wait.
	t.Cleanup(ingestersRing.StopAsync)
```

Reply:
Done in https://github.com/grafana/mimir/pull/7376/commits/15c7d38b1876ea04e5330ad82bfdc89dd744c4b9